### PR TITLE
compat: Refactoring of WrapperRegistration to be reflection based ins…

### PR DESCRIPTION
**Pull request by beckerjohannes**
…tead of hardcoded implementation.

## Pull Request

- [x] I have checked for similar PRs.
- [x] I have read the [contributing guidelines](https://github.com/4Soft-de/harness-model/blob/develop/.github/CONTRIBUTING.md).

### Changes

- [x] Code
- [ ] Documentation
- [ ] Other: 

### Description

This pull request introduces a new mechanism for automatically discovering and registering wrapper classes using a classpath scan, as well as a new annotation to mark wrapper classes. It also refactors the `CompatibilityContext` to allow customization of the default wrapper factory and adds comprehensive tests and fixtures to ensure correct behavior and error handling.

**Automatic Wrapper Registration:**

- Introduced the `@Wraps` annotation for marking wrapper classes as handlers for specific source classes. This annotation is processed at runtime to register wrappers.
- Added the `WrapperAutoRegistrar` utility, which scans a package for `@Wraps`-annotated classes, validates their constructors, and registers them with the `WrapperRegistry`. This supports modular and automatic extension of wrapper functionality.

**CompatibilityContext Enhancements:**

- Refactored `CompatibilityContext` and its builder to accept a customizable `defaultWrapperFactory`. This enables swapping out the default wrapper instantiation logic, making the context more flexible and testable. [[1]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753R31-R33) [[2]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753L45-R53) [[3]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753R96-R97) [[4]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753R122-R127) [[5]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753L125-R138)

**Testing and Fixtures:**

- Added comprehensive tests for the new auto-registration mechanism, including successful registration, handling of wrappers with invalid constructors, and wrappers with empty `@Wraps` values.
- Introduced test fixtures for edge cases: a wrapper with no valid constructor [[1]](diffhunk://#diff-e8ed16c8de225cdb60e2d50133ecab375c147065387f5d3afc721daf5e30e785R1-R44) [[2]](diffhunk://#diff-6ad20c518f7748a5bdf732866d79369a78995e4faf56552c22aa61290e8646d3R1-R29) and a wrapper with an empty `@Wraps` annotation value [[3]](diffhunk://#diff-3c5c47c40d2ef4cd74410fec59638ca33ffa683af5fcc1c1178d8e7353db53b0R1-R39).

These changes significantly improve the extensibility and maintainability of the wrapper registration process, while ensuring robust error handling and configurability.

**References:**  
[[1]](diffhunk://#diff-e19b3c9c9630c10a970927e7d3ca3838cc66f41449c39ce825e487e8a07a99e5R1-R50) [[2]](diffhunk://#diff-5a31829abe800cfe08eba2c7287dffb6a28cf97b4b071ab68789b75a0920400fR1-R119) [[3]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753R31-R33) [[4]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753L45-R53) [[5]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753R96-R97) [[6]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753R122-R127) [[7]](diffhunk://#diff-ec3cb9a64fa2d05f504e10a92f6bad7f6b725d77a0ef13f8acfbb00d0eb78753L125-R138) [[8]](diffhunk://#diff-50de7482043ddd0f9a620877da1fb475289fc803e1658e42675f8306a4e405a0R1-R119) [[9]](diffhunk://#diff-e8ed16c8de225cdb60e2d50133ecab375c147065387f5d3afc721daf5e30e785R1-R44) [[10]](diffhunk://#diff-6ad20c518f7748a5bdf732866d79369a78995e4faf56552c22aa61290e8646d3R1-R29) [[11]](diffhunk://#diff-3c5c47c40d2ef4cd74410fec59638ca33ffa683af5fcc1c1178d8e7353db53b0R1-R39)

### CODE REVIEW

File: compatibility/compatibility-core/src/main/java/com/foursoft/harness/compatibility/core/CompatibilityContext.java [hunk 1]
```diff
@@ -112,6 +119,12 @@ public CompatibilityContextBuilder withUnsupportedMethodCheck(
             return this;
         }
 
+        public CompatibilityContextBuilder withDefaultWrapperFactory(
+                final BiFunction<Context, Object, InvocationHandler> defaultWrapperFactory) {
+            this.defaultWrapperFactory = defaultWrapperFactory;
+            return this;
```
[the remark this answers is not in the document]
Comment (beckerjohannes) [reply compatibility/compatibility-core/src/main/java/com/foursoft/harness/compatibility/core/CompatibilityContext.java:143]: @copilot apply changes based on [this feedback](https://github.com/4Soft-de/harness-model/pull/466#discussion_r3186663261)

File: compatibility/compatibility-core/src/main/java/com/foursoft/harness/compatibility/core/wrapper/WrapperAutoRegistrar.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        int registrations = 0;
+        for (final Class<?> wrapperClass : wrapperClasses) {
+            final Class<?>[] sourceClasses = wrapperClass.getAnnotation(Wraps.class).value();
+            if (sourceClasses.length == 0) {
+                throw new WrapperException(
+                        "@Wraps on " + wrapperClass.getName() + " must declare at least one source class.");
+            }
+            final Constructor<?> ctor = resolveConstructor(wrapperClass, context);
+            for (final Class<?> sourceClass : sourceClasses) {
+                registry.register(sourceClass, target -> instantiate(ctor, context, target));
```
[the remark this answers is not in the document]
Comment (beckerjohannes) [reply compatibility/compatibility-core/src/main/java/com/foursoft/harness/compatibility/core/wrapper/WrapperAutoRegistrar.java:97]: @copilot apply changes based on [this feedback](https://github.com/4Soft-de/harness-model/pull/466#discussion_r3186663247)

[the remark this answers is not in the document]
Comment (beckerjohannes) [reply compatibility/compatibility-core/src/main/java/com/foursoft/harness/compatibility/core/wrapper/WrapperAutoRegistrar.java:97]: @copilot apply changes based on [this feedback](https://github.com/4Soft-de/harness-model/pull/466#discussion_r3186663234)

**Review by takebene — APPROVED** [review]
